src/odemis/driver/test/technolution_test.py

```diff
@@ -0,0 +1,1460 @@
+        mppc = self.MPPC
+
+        # Test over the entire range many options to also let possible floating point error surface
+        for test_repetition in REPEAT_1000:
+            # Check randomly all the options of the dwell time allowed.
+            minimum_dwell_time = scanner.dwellTime.range[0]
+            random_dwell_time = numpy.round(numpy.random.random() * FIX * scanner.dwellTime.range[1], 9)
+            scanner.dwellTime.value = max(random_dwell_time, minimum_dwell_time)
+
+            line_scan_time = scanner.dwellTime.value * mppc.cellCompleteResolution.value[0]
+            remainder_scanning_time = line_scan_time % descanner.clockPeriod.value
+            if remainder_scanning_time is not 0:
+                # Adjusted the flyback time if there is a remainder of scanning time by adding one setpoint to ensure the
+                # line scan time is equal to a equal to a whole multiple of the descan clock period
+                flyback_time = descanner.physicalFlybackTime + (descanner.clockPeriod.value - remainder_scanning_time)
+
+            # Total line scan time is equal to period of the calibration signal, the frequency is the inverse
+            total_line_scan_time = line_scan_time + flyback_time
+
+            calibration_parameters = ASM._assembleCalibrationMetadata()
+
+            # Check types of calibration parameters (output send to the ASM)
+            self.assertIsInstance(calibration_parameters, CalibrationLoopParameters)
+            self.assertIsInstance(calibration_parameters.descan_rotation, float)
+            self.assertIsInstance(calibration_parameters.x_descan_offset, int)
+            self.assertIsInstance(calibration_parameters.y_descan_offset, int)
+            self.assertIsInstance(calibration_parameters.dwell_time, int)
+            self.assertIsInstance(calibration_parameters.scan_rotation, float)
+            self.assertIsInstance(calibration_parameters.x_scan_delay, int)
+            self.assertIsInstance(calibration_parameters.x_scan_offset, float)
+            self.assertIsInstance(calibration_parameters.y_scan_offset, float)
+
+            # Check descan setpoints
+            self.assertIsInstance(calibration_parameters.x_descan_setpoints, list)
+            self.assertIsInstance(calibration_parameters.y_descan_setpoints, list)
+
+            for x_setpoint, y_setpoint in zip(calibration_parameters.x_descan_setpoints,
+                                              calibration_parameters.y_descan_setpoints):
+                self.assertIsInstance(x_setpoint, int)
+                self.assertIsInstance(y_setpoint, int)
+
+            # Check if the time interval used for the descanner setpoints is equal to the descanner clock period
+            x_descan_setpoints_time_interval = total_line_scan_time / len(calibration_parameters.x_descan_setpoints)
+            self.assertEqual(numpy.round(x_descan_setpoints_time_interval, 10) % descanner.clockPeriod.value, 0,
```

Comment:
Because first you need to round and then you can take the modulus almost equal doesn't make sense. Because after rounding you want both to be equal.
If you round after the modulus (so using almost equal like this:
```
            if not almost_equal(x_descan_setpoints_time_interval % descanner.clockPeriod.value, 0, atol=1e-10, rtol=0):
                raise ValueError("Time interval used for the descanner setpoints is not equal to the descanner clock "
                             "period")
```

It will wail with  x_descan_setpoints_time_interval = 9.999999999999999e-06 (which is 1e-05=descan_period when rounded)
So using almost equal in that way doesn't work.